File: PRACTICA PRIMER PARCIAL/Ejercicio16/Python/GenerarSecuencia.py
def generar_secuencia(n):
    secuencia = []
    num = 2

    while len(secuencia) < n:
        if es_primo(num):
            secuencia.append(num)
        num += 1

    return secuencia


def es_primo(num):
    if num < 2:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True


n_terminos = 10
secuencia = generar_secuencia(n_terminos)
print(secuencia)

File: PRACTICA PRIMER PARCIAL/Ejercicio16/Java/GenerarSecuencia.java
import java.util.ArrayList;
import java.util.List;

public class GenerarSecuencia {

    public static List<Integer> generarSecuencia(int n) {
        List<Integer> secuencia = new ArrayList<>();
        int num = 2;

        while (secuencia.size() < n) {
            if (esPrimo(num)) {
                secuencia.add(num);
            }
            num++;
        }

        return secuencia;
    }

    public static boolean esPrimo(int num) {
        if (num < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        int nTerminos = 10;
        List<Integer> secuencia = generarSecuencia(nTerminos);

        for (int num : secuencia) {
            System.out.print(num + " ");
        }
    }
}
